.github/workflows/bot.yml

```
@@ -38,6 +64,11 @@ jobs:
           architecture: x64
       - name: Build Project
         env:
-          SCALA_PROFILE: ${{ matrix.scala }}
-          SPARK_PROFILE: ${{ matrix.spark }}
-        run: mvn install -P "$SCALA_PROFILE,$SPARK_PROFILE" -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+          SCALA_PROFILE: ${{ matrix.scalaProfile }}
+          SPARK_PROFILE: ${{ matrix.sparkProfile }}
+          SPARK_VERSION: ${{ matrix.sparkVersion }}
+          SCALA_BINARY_VERSION: ${{ matrix.scalaBinaryVersion }}
+        run: |
+          mvn install -P "$SCALA_PROFILE,$SPARK_PROFILE" -Dspark.version="$SPARK_VERSION" -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+          ([[ $SPARK_VERSION != 3.2.* ]] && mvn test -P "$SCALA_PROFILE,$SPARK_PROFILE,unit-tests" -Dspark.version="$SPARK_VERSION" -pl hudi-examples/hudi-examples-flink,hudi-examples/hudi-examples-java,hudi-examples/hudi-examples-spark -DfailIfNoTests=false -D "$SPARK_PROFILE" -Dscala.binary.version="$SCALA_BINARY_VERSION")
```

Have you tried put the commands like this to make `SCALA_PROFILE` effective ?

```suggestion
          ([[ $SPARK_VERSION != 3.2.* ]] && mvn test -P unit-tests -Dspark.version="$SPARK_VERSION" -DfailIfNoTests=false -D "$SPARK_PROFILE" -D"$SCALA_PROFILE" -pl hudi-examples/hudi-examples-flink,hudi-examples/hudi-examples-java,hudi-examples/hudi-examples-spark)
```